library/std/src/io/buffered/bufreader/buffer.rs

```diff
@@ -0,0 +1,84 @@
+use crate::cmp;
```

Comment:
I think we should add some module-level documentation for why this module exists (i.e., a safety barrier, primarily).